src/vs/workbench/contrib/codeEditor/browser/languageConfigurationExtensionPoint.ts

```diff
@@ -268,6 +269,30 @@ export class LanguageConfigurationFileHandler {
 		return result;
 	}
 
+	private _extractValidColorizedBracketPairs(languageIdentifier: LanguageIdentifier, configuration: ILanguageConfiguration): CharacterPair[] | null {
+		const source = configuration.colorizedBracketPairs;
+		if (typeof source === 'undefined') {
+			return null;
+		}
+		if (!Array.isArray(source)) {
+			console.warn(`[${languageIdentifier.language}]: language configuration: expected \`colorizedBracketPairs\` to be an array.`);
+			return null;
+		}
+
+		let result: CharacterPair[] | null = null;
+		for (let i = 0, len = source.length; i < len; i++) {
+			const pair = source[i];
+			if (!isCharacterPair(pair)) {
+				console.warn(`[${languageIdentifier.language}]: language configuration: expected \`colorizedBracketPairs[${i}]\` to be an array of two strings.`);
+				continue;
+			}
+			result = result || [];
+			result.push([pair[0], pair[1]]);
+
+		}
+		return result;
+	}
+
```

Comment:
To be honst, I really don't like this.
I can strongly recommend checking out [io-ts](https://github.com/gcanti/io-ts/blob/master/index.md#typescript-integration) or get inspired by my own library [semantic json](https://github.com/hediet/semantic-json#basic-usage).

Being able to specify deserialization logic, validation logic, json schema and the defaults in a single place (descriptively) is a huge deal. Not only here, but also for all settings.

Reply:
cc @aeschli -- would be nice if we could validate an object against a schema programmatically.